deploy/firesim

```diff
@@ -171,20 +171,20 @@ def terminaterunfarm(runtime_conf, terminatesomef1_16, terminatesomef1_4, termin
     runtime_conf.terminate_run_farm(terminatesomef1_16, terminatesomef1_4, terminatesomef1_2,
                                     terminatesomem4_16, forceterminate)
 
-def shareagfi(build_config):
+def shareagfi(build_config_file: BuildConfigFile) -> None:
```

Comment:
Not saying that we should change this, only expressing my own opinion here.  For methods like this that take a 'config', now that we have the type-hints in place, it feels very redundant to name the param a reformatted version of the type.  That probably comes from the C++ coding standard I'm used to where we don't want type changes to have to ripple through the code.  I think going forward that `config` would be a fine name here where the method is super short and the type removes any ambiguity.

Reply:
I always thought this was a bit verbose, now that there are types I like a shorter name.

Reply:
I'll punt this also to a later PR.